dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/ContentBody.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.OptionalLong;
+
+public interface ContentBody extends Closeable {
+
+    /** The content of this request body, possibly empty. */
+    void writeTo(OutputStream output) throws IOException;
+
+    /** A HTTP content type (e.g., "application/json") indicating the type of content. */
+    String contentType();
```

Comment:
Ok, so another quirk of apache implementation of this, is that #isRepeatable is based on getContentLength() != 1; interestingly we have a an internal client that deals with InputStreams that it may arbitrarily stream to disk, to make sure it can get retries. But I think that might be ameanable to providing the contentLength of those values, to get repeatability.

Reply:
We don't currently provide repeatability for `BinaryRequestBody` binary requests, it's reasonable not to support it in this case when lengths aren't provided -- I would not want to record contents to disk, as disk performance can be finicky, and space may not be available.

Reply:
Yeah, I agree it's reasonable. Internal client likely was written because it was dealing with finnicky upstreams, so the way it deals with that is write to disk first, then send to itself. Anyway, feels like we can support what they need, so shouldn't be a big deal.